base/test.jl

```diff
@@ -220,6 +228,60 @@ function do_test(result::ExecutionResult, orig_expr)
     record(get_testset(), testres)
 end
 
+# @test_fail_expected - check if the expression evaluates to false
+# In the special case of a comparison, e.g. x == 5, generate code to
+# evaluate each term in the comparison individually so the results
+# can be displayed nicely.
+"""
+    @test_fail_expected ex
+
+Tests that the expression `ex` evaluates to `false`.
+Returns a `Pass` `Result` if it does, a `Fail` `Result` if it is
+`true`, and an `Error` `Result` if it could not be evaluated.
+"""
+macro test_fail_expected(ex)
```

Comment:
It's not good to copy and paste this much code. This could be implemented as something like

```
macro test_fail_expected(ex)
    :(@test !($ex))
end
```

or by calling a common helper function.